components/domains/steps/upsellCard.tsx

```diff
@@ -1,19 +1,41 @@
 import styles from "../../../styles/components/upsellCard.module.css";
-import React, { FunctionComponent } from "react";
+import React, { FunctionComponent, useEffect, useState } from "react";
 import { FormControlLabel, Radio, RadioGroup } from "@mui/material";
 import textFieldStyles from "../../../styles/components/textField.module.css";
+import { CurrencyType } from "@/utils/constants";
 
 type UpsellCardProps = {
   upsellData: Upsell;
   enabled: boolean;
   onUpsellChoice: (isUpselled: boolean) => void;
+  invalidBalance: boolean;
+  displayedCurrency: CurrencyType;
 };
 
 const UpsellCard: FunctionComponent<UpsellCardProps> = ({
   upsellData,
   enabled,
   onUpsellChoice,
+  invalidBalance,
+  displayedCurrency,
 }) => {
+  const [isUserChoice, setIsUserChoice] = useState<boolean>(false);
+
+  useEffect(() => {
+    setIsUserChoice(false);
+  }, [displayedCurrency]);
```

Comment:
Using displayed currency is not in my opinion a good way to do it. 

Imo you should simply say, that by default the discount is on but if `invalidBalance` is true then you put it off. Then you manage the manual change on top of that.